openmls/src/group/tests/test_commit_validation.rs

```diff
@@ -226,217 +251,172 @@ fn test_valsem200(ciphersuite: Ciphersuite, backend: &impl OpenMlsCryptoProvider
+            .unwrap();
+        let psk_id = PreSharedKeyId::new(
+            ciphersuite,
+            backend.rand(),
+            Psk::External(ExternalPsk::new(rand)),
+        )
+        .unwrap();
+        let psk_key = psk_id.tls_serialize_detached().unwrap();
+        backend.key_store().store(&psk_key, &psk_bundle).unwrap();
+        queued(Proposal::PreSharedKey(PreSharedKeyProposal::new(psk_id)))
+    };
 
-    let err = charlie_group
-        .process_unverified_message(unverified_message, None, backend)
-        .expect_err("Could process unverified message despite missing path.");
+    let update_proposal = || {
+        let key_package = alice_group
+            .member(alice_group.own_leaf_index())
+            .unwrap()
+            .clone();
+        queued(Proposal::Update(UpdateProposal { key_package }))
+    };
 
-    assert_eq!(
-        err,
-        UnverifiedMessageError::InvalidCommit(StageCommitError::RequiredPathNotFound)
-    );
+    let remove_proposal = || {
+        queued(Proposal::Remove(RemoveProposal {
+            removed: charlie_group.own_leaf_index(),
+        }))
+    };
 
-    let original_update_plaintext =
-        VerifiableMlsAuthContent::tls_deserialize(&mut serialized_update.as_slice())
-            .expect("Could not deserialize message.");
+    let gce_proposal = || {
+        queued(Proposal::GroupContextExtensions(
+            GroupContextExtensionProposal::new(alice_group.group().group_context_extensions()),
+        ))
+    };
 
-    // Positive case
-    let unverified_message = charlie_group
-        .parse_message(MlsMessageIn::from(original_update_plaintext), backend)
-        .expect("Could not parse message.");
-    charlie_group
-        .process_unverified_message(unverified_message, None, backend)
-        .expect("Unexpected error.");
+    // ExternalInit Proposal cannot be used alone and has to be in an external commit which
+    // always contains a path anyway
+    // TODO: #916 when/if AppAck proposal are implemented (path not required)
+    // TODO: #751 when ReInit proposal validation are implemented (path not required). Currently one
+    // cannot distinguish when the commit has a single ReInit proposal from the commit without proposals
+    // in [CoreGroup::apply_proposals()]
+    let cases = vec![
+        (vec![add_proposal()], false),
+        (vec![psk_proposal()], false),
+        (vec![update_proposal()], true),
+        (vec![remove_proposal()], true),
+        (vec![gce_proposal()], true),
+        // !path_required + !path_required = !path_required
+        (vec![add_proposal(), psk_proposal()], false),
+        // path_required + !path_required = path_required
+        (vec![remove_proposal(), add_proposal()], true),
+        // path_required + path_required = path_required
+        (vec![update_proposal(), remove_proposal()], true),
+        // TODO: this should work if GCE proposals were implemented
+        // (vec![add_proposal(), gce_proposal()], true),
```

Comment:
Should this be commented in? If not, the TODO should probably link to an issue.

Reply:
I agree, it should link to issue #566. If you agree and dare reopening it, I'll link it in the TODO